src/sparse/serde_traits.rs

```diff
@@ -1,6 +1,33 @@
 use super::*;
-pub(crate) use serde::{Deserialize, Serialize};
+pub(crate) use serde::ser::SerializeStruct;
+pub(crate) use serde::{Deserialize, Serialize, Serializer};
 use std::convert::TryFrom;
+use std::ops::Deref;
+
+impl<N, I: SpIndex, Iptr: SpIndex, IptrStorage, IStorage, DStorage> Serialize
+    for CsMatBase<N, I, IptrStorage, IStorage, DStorage, Iptr>
+where
+    Iptr: Serialize,
+    I: Serialize,
+    N: Serialize,
+    IptrStorage: Deref<Target = [Iptr]>,
+    IStorage: Deref<Target = [I]>,
+    DStorage: Deref<Target = [N]>,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("CsMatBase", 3)?;
```

Comment:
Should this be `6` instead of `3`?

Reply:
Oops I don't know how that slipped through. It's fixed now.